onnxscript/type_annotation.py

```diff
@@ -62,9 +69,41 @@ def is_attr(pytype: type):
 
 
 def is_tensor(typeinfo):
-    return hasattr(typeinfo, "to_type_proto")
+    try:
+        return isinstance(typeinfo, onnxscript.onnx_types.TensorType) or issubclass(
+            typeinfo, onnxscript.onnx_types.TensorType
+        )
+    except:
+        print("Opps")
+    # return hasattr(typeinfo, "to_type_proto")
     # return isinstance(typeinfo, onnxscript.Tensor)  # TODO
 
 
 def is_valid(typeinfo):
     return is_attr(typeinfo) or is_tensor(typeinfo)
+
+
+def validate(typeinfo):
+    if not is_valid(typeinfo):
+        raise ValueError(f"Unsupported type annotation {typeinfo}")
+
+
+def get_return_types(typeinfo: type | typing.Sequence[type]) -> typing.Sequence[type]:
```

Comment:
```suggestion
def get_return_types(typeinfo: type | Sequence[type]) -> Sequence[type]:
```

nit

Reply:
Currently importing only `typing` in this file (due to 3.8 and 3.7 version differences for get_origin/get_args).